Resfeber/CoreData/TripService.swift

```diff
@@ -12,7 +12,7 @@ import CoreData
 public final class TripService {
     // MARK: - Properties
     let managedObjectContext: NSManagedObjectContext
-    let coreDataStack: CoreDataStack
+    var coreDataStack = CoreDataStack.sharedCoreDataStack
```

Comment:
I would just get rid of this entirely and have a property for `let context = CoreDataStack.shared.mainContext` because really the context is what you are wanting right? If you're wanting a background context at some point then you can just simply create one in the scope of your method `let bgContext = CoreDataStack.shared.newDerivedContext` or whatever you guys called it.